packages/sdk/src/protocols/peer_manager.ts

```diff
@@ -0,0 +1,113 @@
+import { Peer, PeerId } from "@libp2p/interface";
+import { ConnectionManager, getHealthManager } from "@waku/core";
+import { BaseProtocol } from "@waku/core/lib/base_protocol";
+import { IHealthManager } from "@waku/interfaces";
+import { Logger } from "@waku/utils";
+import { Mutex } from "async-mutex";
+
+export class PeerManager {
+  private peers: Map<string, Peer> = new Map();
+  private healthManager: IHealthManager;
+
+  private readMutex = new Mutex();
+  private writeMutex = new Mutex();
+  private writeLockHolder: string | null = null;
+
+  public constructor(
+    private readonly connectionManager: ConnectionManager,
+    private readonly core: BaseProtocol,
+    private readonly log: Logger
+  ) {
+    this.healthManager = getHealthManager();
+    this.healthManager.updateProtocolHealth(this.core.multicodec, 0);
+  }
+
+  public getWriteLockHolder(): string | null {
```

Comment:
it should be private

Reply:
Since this isn't a public interface, it's helpful to have this function be `public` to help debug any deadlocks